libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,662 @@
+        this.navigationIndicator = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation-indication');
+    }
+
+    /** setter and getter for hasByLine*/
+    @Input('hasByLine')
+    get hasByLineValue(): boolean {
+        return this.hasByLine;
+    }
+
+    set hasByLineValue(value: boolean) {
+        this.hasByLine = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--byline');
+    }
+
+    /** setter and getter for radio button and checkbox*/
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+
+    /** @hidden */
+    /**handline keyboard operations
+    * in template on list and list items
+    */
+    handleKeyDown(event: KeyboardEvent): boolean {
+        event.stopImmediatePropagation();
+        if (this.keyManager) {
+            if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+                return false;
+            } else if (event.keyCode === ENTER || event.keyCode === SPACE) {
+                this.updateNavigation(event);
+                return false;
+            }
+        }
+    }
+
+    /**@hidden
+     * binded in template on scroll
+     */
+    scrollHandler(): void {
+        if (!this.loading && this.loadOnScroll) {
+            this.getMoreData();
+        }
+    }
+
+    /**@hidden
+   *  Handles lazy loading data
+   * used in template
+   * onscroll and on more button click
+   */
+    getMoreData(): void {
+        this.loading = true;
+        of(this._loadNewItems())
+            .pipe(
+                tap(data => {
+                    if (data === null || data === undefined) {
+                        console.error('===Invalid Response recived===');
+                    }
+                }),
+                delay(this.delayTime)
+            ).subscribe(result => {
+                if (result !== null && result !== undefined) {
+                    for (let i = this.items.length, j = 0; j < result.length; ++i, ++j) {
+                        this.items[i] = result[j];
+                    }
+                }
+                this.loading = false;
+                this.stateChanges.next(this.items);
+                this._changeDetectorRef.markForCheck();
+
+            });
+    }
+
+    /**@hidden
+ *  used in tempate to get Selected items from a list
+ * event:any to avoid code duplication**/
+    onSelectionChanged(event: any): void {
+        if (event.target.checked) {
+            this.selectionModel.select(event.target.parentNode.parentNode.parentNode);
+        } else {
+            this.selectionModel.deselect(event.target.parentNode.parentNode.parentNode);
+        }
+    }
+
+    /** @hidden */
+    /**  on Update navgiation styles for non navigated items
+     * event:any to avoid code duplication**/
+    @HostListener('click', ['$event'])
+    updateNavigation(event: any): void {
```

Comment:
This is method as well. Even it has decorators, Again on top of constructor should be only properties.

Reply:
done